.github/workflows/release-publish.yaml

```
@@ -0,0 +1,56 @@
+name: Release
+on:
+  push:
+    tags:
+      - v*
+
+permissions:
+  contents: write
+
+jobs:
+  source-wheel:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+          cache: "pip"
+          check-latest: true
+      - name: Build packages
+        run: |
+          pip install build twine
+          python3 -m build
+          twine check dist/*
+      - name: Upload pypi packages artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: pypi-packages
+          path: dist
+  pypi:
+    needs: source-wheel
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get packages
+        uses: actions/download-artifact@v3
+        with:
+          name: pypi-packages
```

Copying over from craft-archives:
```suggestion
          name: pypi-packages
          path: dist/
```